src/login/logind-dbus.c

```diff
@@ -2038,6 +2042,68 @@ static usec_t nologin_timeout_usec(usec_t elapse) {
         return LESS_BY(elapse, 5 * USEC_PER_MINUTE);
 }
 
+int manager_load_scheduled_shutdown(Manager *m) {
+        _cleanup_fclose_ FILE *f = NULL;
+        int r;
+
+        f = fopen(SHUTDOWN_SCHEDULE_FILE, "r");
+        if (!f) {
+                if (errno == ENOENT)
+                        return 0;
+                return -errno;
+        }
+
+        for (;;) {
+                _cleanup_free_ char *line = NULL;
+                const char *val, *l;
+
+                r = read_line(f, LONG_LINE_MAX, &line);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read serialization line: %m");
+                if (r == 0)
+                        break;
+
+                l = strstrip(line);
+                if (isempty(l)) /* end marker */
+                        break;
```

Comment:
Rather than `break;`, I think it is better to `continue;` the loop. As it may contains blank lines.

Reply:
doesn't really matter I guess. logind itself would not produce empty lines creating the file